api/base/middleware.py

```diff
@@ -139,3 +140,19 @@ def process_response(self, request, response):
             response.content = s.getvalue()
 
         return response
+
+
+class SloanIdMiddleware(MiddlewareMixin):
+    """Sloan middleware give all users a unique id, logged in or not."""
+
+    def process_response(self, request, response):
+        """give user a Sloan ID if they don't have one already"""
+        if not request.COOKIES.get(settings.SLOAN_ID_COOKIE_NAME):
```

Comment:
hm it's confusing that this file imports api settings twice in two different ways -- any reason this is using `settings` (`django.conf.settings`) instead of `api_settings` (`api.base.settings`), or is it just a coin flip that doesn't matter?

Reply:
Yes I tried fixing this, but it's necessary because the CORs middleware needs to load the Whitelist of Origins before the settings file can be totally evaluated. We should probably make a ticket to unify the settings file and stop some of this weird stuff.